### PR TITLE
DEVENGAGE-1610 Axios proxy bug

### PR DESCRIPTION
**Pull request by dginty4**

### CODE REVIEW

File: resources/sdk/purecloudjavascript/templates/ApiClient.mustache [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { default as axios } from 'axios';
 import Configuration from './configuration.js';
 import { default as qs } from 'qs';
+const { HttpsProxyAgent } = require('hpagent')
```
Comment (dginty4) [inline resources/sdk/purecloudjavascript/templates/ApiClient.mustache:4]: I couldn't get the import to work like the other imports, anybody know how I could?

File: resources/sdk/purecloudjavascript/templates/ApiClient.mustache [hunk 2]
```diff
@@ -275,16 +276,21 @@ class ApiClient {
...
-
 			const headers = {
 				'Authorization': `Basic ${authHeader}`
 			}
+            let httpsAgent;
+            if (this.proxy !== undefined){
+                httpsAgent = new HttpsProxyAgent({
+                    proxy : `${this.proxy.protocol}://${this.proxy.host}:${this.proxy.port}`
+                });
+            }
```
Comment (dginty4) [inline resources/sdk/purecloudjavascript/templates/ApiClient.mustache:287]: I also can't seem to get the formatting right, it looks right on any editor I open it on except here

**Review by timsmithgenesys — COMMENTED** [review]
Did we get a permanent proxy set up to be able to add tests? If so, tests should be added. If not, tests should be added when the proxy is deployed and accessible during the build.

**Review by carnellj-genesys — CHANGES_REQUESTED** [review]
1.  Didn't have any technical comments besides what Tim pointed out.
2. I don't see any tests here.  I would like to see a test case using the target proxy.  While manual tests are good, automated tests will help us find stuff later if somebody changes something again.

Nice work on this Declan.

File: resources/sdk/purecloudjavascript/templates/ApiClient.mustache (hunk 1 above)
Comment (dginty4) [reply resources/sdk/purecloudjavascript/templates/ApiClient.mustache:4]: Thanks Tim. `import { HttpsProxyAgent } from 'hpagent'` doesn't seem to work so I'll have to leave it as is

**Comment by dginty4** [comment]
Would someone be able to build the sdk local? I think the security group for the proxy servers EC2 instance might only allow my IP through but I'm not 100%

**Review by carnellj-genesys — APPROVED** [review]

**Pull request merged**